zebra-chain/src/sapling/spend.rs

```diff
@@ -134,6 +134,20 @@ impl ZcashSerialize for Spend<SharedAnchor> {
     }
 }
 
+impl ZcashDeserialize for Spend<SharedAnchor> {
```

Comment:
The spec puts some parts of the transaction v5 spend in different arrays.

It might be easier to serialise and deserialise all the `Spend<SharedAnchor>`s as part of the `ShieldedData<SharedAnchor>`.

Please reference the relevant spec section in a doc comment for the serialise and deserialise impls.

Reply:
Let me know if you'd like to work through this change together - it's going to be a bit more complicated because of the TrustedPreallocate tests.